fix: Small fixes resolving naming and nil-pointers

nderjung opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

  Kraftkit follows the same guidelines as the Unikraft Open Source Project.

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

GitHub-Fixes: https://github.com/unikraft/kraftkit/issues/1551

nderjung committed:
feat: Rename "elfloader" to "runtime"

This makes it consistent with how it is presented externally, i.e.
via the `runtime` element in the `Kraftfile`.

Signed-off-by: Alexander Jung <alex@unikraft.io>

nderjung committed:
fix(runtime): Fix possible nil-pointer exception

GitHub-Fixes: #1551
Signed-off-by: Alexander Jung <alex@unikraft.io>

LucaSeri reviewed (commented):
Hi @nderjung. Thanks for this. ``kraft pkg`` works as expected now.
One question about the ``runtime`` flag: is the following setup meant to work?

```
Kraftfile
spec: v0.6

rootfs: ./Dockerfile
  
cmd: ["/server"]
```
then:
```
kraft run --runtime unikraft.org/base:latest .
```
I get the following errors:
```
 E  could not determine how to run provided input: first positional argument is a directory: .
first positional argument is a directory: /home/kero/unikraft-work/catalog/examples/http-go1.21
cannot run project build without unikraft
cannot run project without runtime directive
arguments represent a file or directory
arguments represent a file or directory
```

Perhaps the runtime flag should also be injected elsewhere?

nderjung committed:
fix: Explicitly and fully propagate use of `--runtime` flag

The `--runtime` flag for `kraft run` was implicitly registered by
the `runtime` package.  In this commit we remove this implicit
registration and declare it properly as `RunOptions` attribute.
As a result, we explicitly use it also in relevant runners which
employ a runtime name.

Signed-off-by: Alexander Jung <alex@unikraft.io>

nderjung commented:
Thanks @LucaSeri, I've broken up the commits to be a bit more explicit about the (re)introduction of the `--runtime` flag.  Additionally, I've addressed the issue that surrounds not having a `runtime` element in a `Kraftfile` as this flag was not properly used in the respective `runner`.

LucaSeri reviewed (approved):
Thanks! Works well.
